kotlin-desktop-toolkit-win32/build.gradle.kts

```diff
@@ -112,6 +115,34 @@ val downloadJExtractTask = tasks.register<DownloadJExtractTask>("downloadJExtrac
     jextractDirectory = layout.buildDirectory.dir("jextract")
 }
 
+val downloadAngleTaskByPlatform = enabledPlatforms.associateWith { platform ->
+    tasks.register<DownloadAngleTask>("downloadAngle-${buildPlatformRustTarget(platform)}") {
+        this.platform = platform
+        version = "7fea539cc9"
+        outputDirectory = layout.buildDirectory.dir("angle-${angleArch(platform.arch)}")
+    }
+}
+
+val collectNativeArtifactsTaskByTarget = buildMap {
+    for (platform in enabledPlatforms) {
+        val downloadAngleTask = downloadAngleTaskByPlatform[platform]!!
+        for (profile in profiles) {
+            val buildNativeTask = compileNativeTaskByTarget[RustTarget(platform, profile)]!!
+            val collectWindowsArtifactsTask = tasks.register<CollectWindowsArtifactsTask>(
```

Comment:
You should not need a task for that.

Could you try with:
```kotlin
put(RustTarget(platform, profile), project.objects.fileCollection().from(
    downloadAngleTask.flatMap { it.binaries },
    buildNativeTask.flatMap { it.libraryFile },
).builtBy(downloadAngleTask, buildNativeTask))
```
and remove completely `CollectWindowsArtifactsTask` class, and registration of tasks?

Reply:
Nope, that doesn't work for us, because we add an artifact to a consumable, and that thing only accepts the following:

```
* What went wrong:
Cannot convert the provided notation to an object of type ConfigurablePublishArtifact: file collection.
The following types/formats are supported:
  - Instances of ConfigurablePublishArtifact.
  - Instances of PublishArtifact.
  - Instances of AbstractArchiveTask, for example jar.
  - Instances of Provider<RegularFile>.
  - Instances of Provider<Directory>.
  - Instances of Provider<File>.
  - Instances of RegularFile.
  - Instances of Directory.
  - Instances of File.
  - Maps with 'file' key
```

Reply:
Yes you can, I just found out about it today finally, use:

```
nativeConsumable.configure {
  outgoing.artifacts(<your file collection here>) {
    builtBy(<eventually your tasks here>)
  }
}
```

Reply:
Just a follow-up here for history purposes, turns out, it won't work due to that: https://github.com/gradle/gradle/issues/34650